### PR TITLE
Fix construct_grid TypeError, add tests

### LINKED ISSUE #33: TypeError in `dependence`
In `dependence.py`:

`if isinstance(grid_values, Union[List, Tuple, np.ndarray]):`

raises: `TypeError: Subscripted generics cannot be used with class and instance checks`

### PR DESCRIPTION
**Pull request by jack-fs**
This appears to fix #33. I've added a simple test for construct grid which should help us catch any regressions.

### CODE REVIEW

File: cinspect/dependence.py [hunk 1]
```diff
@@ -139,7 +140,12 @@ def construct_grid(
     grid_counts = None
     grid = None
 
-    if isinstance(grid_values, Union[List, Tuple, np.ndarray]):
+    def is_list_tuple_or_array(
+        x,
+    ):  # For Python >3.10:  -> TypeGuard[Union[List, Tuple, np.ndarray]]:
+        return isinstance(x, List) or isinstance(x, Tuple) or isinstance(x, np.ndarray)
```
Comment (dsteinberg) [inline cinspect/dependence.py:146]: you could also do this: `isinstance(x, (List, Tuple, np.ndarray))`

See: https://www.w3schools.com/python/ref_func_isinstance.asp

Then you may not have to make this into a function

File: tests/test_dependence.py [hunk 2]
```diff
@@ -75,3 +80,36 @@ def ice_fn(Xb, yb):
...
+            n_unique = len(np.unique(v))
+            if n_unique > auto_threshold:
+                assert counts is None, f"expected counts to be None but it was {counts}"
+            else:
+                assert counts is not None
+        else:
+            raise ValueError(f"Unexpected grid_values argument {grid_values}")
+    else:
+        assert counts is None, f"expected counts to be None, but it was {counts}"
+    assert isinstance(grid, np.ndarray), "construct_grid constructed grid is invalid"
```
Comment (dsteinberg) [inline tests/test_dependence.py:115]: Awesome! Thanks for doing this, I wasn't looking forward to it ;-)

**Review by dsteinberg — APPROVED** [review]
Looks good!

I added a comment in - up to you whether you want to change the code accordingly. When you are ready, merge away!

File: cinspect/dependence.py (hunk 1 above)
Comment (jack-fs) [reply cinspect/dependence.py]: Oh well that's much nicer (and presumably gets `mypy` type narrowing support for free as well)

**Pull request merged**